src/formats/jpeg/Frame.zig

```diff
@@ -82,145 +86,181 @@ pub fn deinit(self: *Self) void {
-            const cb_sampled_y = (mcu_y * self.frame_header.components[1].vertical_sampling_factor) / max_vertical_sampling_factor;
-            const cr_sampled_y = (mcu_y * self.frame_header.components[2].vertical_sampling_factor) / max_vertical_sampling_factor;
-
-            // y coordinates of each component in the block
-            const y_block_y = y_sampled_y % 8;
-            const cb_block_y = cb_sampled_y % 8;
-            const cr_block_y = cr_sampled_y % 8;
-
-            for (0..mcu_width) |mcu_x| {
-                const x = mcu_origin_x + mcu_x;
-                if (x >= width) continue;
-
-                // x coordinates of each component applied to the sampling factor
-                const y_sampled_x = (mcu_x * self.frame_header.components[0].horizontal_sampling_factor) / max_horizontal_sampling_factor;
-                const cb_sampled_x = (mcu_x * self.frame_header.components[1].horizontal_sampling_factor) / max_horizontal_sampling_factor;
-                const cr_sampled_x = (mcu_x * self.frame_header.components[2].horizontal_sampling_factor) / max_horizontal_sampling_factor;
-
-                // x coordinates of each component in the block
-                const y_block_x = y_sampled_x % 8;
-                const cb_block_x = cb_sampled_x % 8;
-                const cr_block_x = cr_sampled_x % 8;
-
-                const y_block_ind = (y_sampled_y / 8) * self.frame_header.components[0].horizontal_sampling_factor + (y_sampled_x / 8);
-                const cb_block_ind = (cb_sampled_y / 8) * self.frame_header.components[1].horizontal_sampling_factor + (cb_sampled_x / 8);
-                const cr_block_ind = (cr_sampled_y / 8) * self.frame_header.components[2].horizontal_sampling_factor + (cr_sampled_x / 8);
-
-                const mcu_Y = &self.mcu_storage[mcu_id][0][y_block_ind];
-                const mcu_Cb = &self.mcu_storage[mcu_id][1][cb_block_ind];
-                const mcu_Cr = &self.mcu_storage[mcu_id][2][cr_block_ind];
-
-                const Y: f32 = @floatFromInt(mcu_Y[y_block_y * 8 + y_block_x]);
-                const Cb: f32 = @floatFromInt(mcu_Cb[cb_block_y * 8 + cb_block_x]);
-                const Cr: f32 = @floatFromInt(mcu_Cr[cr_block_y * 8 + cr_block_x]);
-
-                const Co_red = 0.299;
-                const Co_green = 0.587;
-                const Co_blue = 0.114;
-
-                const r = Cr * (2 - 2 * Co_red) + Y;
-                const b = Cb * (2 - 2 * Co_blue) + Y;
-                const g = (Y - Co_blue * b - Co_red * r) / Co_green;
-
-                pixels[y * width + x] = .{
-                    .r = @intFromFloat(std.math.clamp(r + 128.0, 0.0, 255.0)),
-                    .g = @intFromFloat(std.math.clamp(g + 128.0, 0.0, 255.0)),
-                    .b = @intFromFloat(std.math.clamp(b + 128.0, 0.0, 255.0)),
-                };
+pub fn YCbCrToRgbBlock(self: *Self, y_block: *[3]Block, cbcr_block: *[3]Block, v: usize, h: usize) void {
```

Comment:
This should start with a lowercase letter, in Zig functions that starts with an uppercase letter are for generics by convention